python/ray/train/tests/test_base_trainer.py

```diff
@@ -204,38 +204,66 @@ def __init__(self):
         def warning(self, msg):
             self.warnings.append(msg)
 
+        def warn(self, msg, **kwargs):
+            self.warnings.append(msg)
+
         def info(self, msg):
             print(msg)
 
     try:
-        old = base_trainer.logger
-        base_trainer.logger = MockLogger()
+        old = tuner_internal.warnings
+        tuner_internal.warnings = MockLogger()
 
         # Fraction correctly specified.
-        DummyTrainer(
+        trainer = DummyTrainer(
             train_loop,
             scaling_config=ScalingConfig(num_workers=1, _max_cpu_fraction_per_node=0.9),
             datasets={"train": ray.data.range(10)},
         )
-        assert not base_trainer.logger.warnings
+        trainer.fit()
+        assert not tuner_internal.warnings.warnings
 
         # No datasets, no fraction.
-        DummyTrainer(
+        trainer = DummyTrainer(
             train_loop,
             scaling_config=ScalingConfig(num_workers=1),
         )
-        assert not base_trainer.logger.warnings
+        trainer.fit()
+        assert not tuner_internal.warnings.warnings
 
         # Should warn.
-        DummyTrainer(
+        trainer = DummyTrainer(
+            train_loop,
+            scaling_config=ScalingConfig(num_workers=3),
+            datasets={"train": ray.data.range(10)},
+        )
+        trainer.fit()
+        assert (
+            len(tuner_internal.warnings.warnings) == 1
+        ), tuner_internal.warnings.warnings
+        assert "_max_cpu_fraction_per_node" in tuner_internal.warnings.warnings[0]
+
+        # Warn if num_samples is configured
+        trainer = DummyTrainer(
             train_loop,
             scaling_config=ScalingConfig(num_workers=1),
             datasets={"train": ray.data.range(10)},
         )
-        assert len(base_trainer.logger.warnings) == 1, base_trainer.logger.warnings
-        assert "_max_cpu_fraction_per_node" in base_trainer.logger.warnings[0]
+        tuner = tune.Tuner(trainer, tune_config=tune.TuneConfig(num_samples=3))
+        tuner.fit()
+        assert (
+            len(tuner_internal.warnings.warnings) == 2
+        ), tuner_internal.warnings.warnings
```

Comment:
Add `assert "_max_cpu_fraction_per_node" in tuner_internal.warnings.warnings[0]`?